hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcServer.java

```diff
@@ -726,9 +727,10 @@ static String getMethodName() {
    * @return List of name spaces in the federation on
    * removing the already invoked namespaceinfo.
    */
-  private Set<FederationNamespaceInfo> getNameSpaceInfo(String nsId) {
+  private Set<FederationNamespaceInfo> getNameSpaceInfo(
+      Set<FederationNamespaceInfo> nss, String nsId) {
     Set<FederationNamespaceInfo> namespaceInfos = new HashSet<>();
-    for (FederationNamespaceInfo ns : namespaceInfos) {
+    for (FederationNamespaceInfo ns : nss) {
```

Comment:
Very good catch.

Catching the issue with the log is fine but I think there must be an easier way to test that the feature works.
Is there any other unit test we can add?

Reply:
Thanks @goiri for your comments and suggestion. I rolled back the way catching with log and replaced with calling #getServerDefaults (create a separate unit test #testInvokeAtAvailableNs). PTAL. Thanks a lot.